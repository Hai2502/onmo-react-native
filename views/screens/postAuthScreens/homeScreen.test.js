import React from 'react';
import Enzyme from 'enzyme';
import renderer from 'react-test-renderer';
import HomeScreen from './homeScreen';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallow, configure } from 'enzyme';
Enzyme.configure({ adapter: new Adapter() });

describe('HomeScreen calls userSignout', () =>{
  it('renders correctly', async () => {
    renderer.create(<HomeScreen />);
   });
   it('should call onPress', () => {
    const mockFunc = jest.fn();
    const component = shallow(<HomeScreen onPress={mockFunc} />);    
    component.dive().simulate('press');
    expect(mockFunc).toHaveBeenCalled();
  });

})
