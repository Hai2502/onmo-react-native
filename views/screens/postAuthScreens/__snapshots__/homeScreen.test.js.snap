// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen calls userSignout renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Now you're authenticated! Welcome!
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        LOG OUT
      </Text>
    </View>
  </View>
</View>
`;

exports[`rerender correctly 1`] = `[Function]`;
